src/cockpit/beiboot.py

```diff
@@ -212,60 +245,68 @@ def __init__(self, router: Router, destination: str, args: argparse.Namespace):
         super().__init__(router)
 
     async def do_connect_transport(self) -> None:
-        # do we have user/password (Basic auth) from the login page?
-        auth = await self.router.request_authorization("*")
-        ssh_opts = []
+        # Choose your own adventure...
+        if os.path.exists('/.flatpak-info'):
+            await self.connect_from_flatpak()
+        else:
+            await self.connect_from_bastion_host()
+
+    async def connect_from_flatpak(self) -> None:
+        # We want to run a python interpreter somewhere...
+        cmd, env = python_interpreter('cockpit-bridge')
+
+        # Remote host?  Wrap command with SSH
+        if self.destination != 'localhost':
+            # we run ssh and thus the helper on the host, always use the xdg-cache helper
+            cmd, env = via_ssh(cmd, self.destination, ensure_ferny_askpass())
```

Comment:
... and moreover, it would stop making this look like it'd entirely ignore the env from the python_interpreter() call above.